sphinx/screens-detail/tribe-members-list/tribe-members-list/src/main/java/chat/sphinx/tribe_members_list/ui/adapter/TribeMembersListAdapter.kt

```diff
@@ -0,0 +1,310 @@
+            } catch (e: IndexOutOfBoundsException) {
+                false
+            }
+
+            if (sameList) {
+                sameList = same
+            }
+
+            return same
+        }
+
+    }
+
+    private val tribeMembers = ArrayList<TribeMemberHolderViewState>(viewModel.currentViewState.list)
+
+    override fun onStart(owner: LifecycleOwner) {
+        super.onStart(owner)
+
+        onStopSupervisor.scope.launch(viewModel.mainImmediate) {
+            viewModel.collectViewState { viewState ->
+                if (viewState is TribeMembersListViewState.ListMode) {
+                    if (viewState.firstPage) {
+                        tribeMembers.clear()
+                    }
+
+                    if (!viewState.loading && viewState.list.isNotEmpty()) {
+                        if (tribeMembers.isEmpty()) {
+                            tribeMembers.addAll(viewState.list)
+                        } else {
+                            val diff = Diff(tribeMembers, viewState.list)
+
+                            withContext(viewModel.dispatchers.default) {
+                                DiffUtil.calculateDiff(diff)
+                            }.let {
+                                if (!diff.sameList) {
+                                    tribeMembers.removeLast()
+                                    tribeMembers.addAll(viewState.list)
+                                }
+                            }
+                        }
+                        this@TribeMembersListAdapter.notifyDataSetChanged()
+                    }
+                }
+            }
+        }
+    }
+
+    override fun getItemCount(): Int {
+        return tribeMembers.size
+    }
+
+    fun getPendingTribeMembersCount(): Int {
+        return tribeMembers.count { tribeMember -> tribeMember.contactDto?.pending == true }
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TribeMemberViewHolder {
+        val binding = LayoutTribeMemberHolderBinding.inflate(
+            LayoutInflater.from(parent.context),
+            parent,
+            false
+        )
+
+        return TribeMemberViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(holder: TribeMemberViewHolder, position: Int) {
+        holder.bind(position)
+    }
+
+    inner class TribeMemberViewHolder(
+        private val binding: LayoutTribeMemberHolderBinding
+    ): RecyclerView.ViewHolder(binding.root) {
+
+        private val holderJobs: ArrayList<Job> = ArrayList(6)
+        private var disposable: Disposable? = null
+        private var tribeMemberHolderViewState: TribeMemberHolderViewState? = null
+
+        fun bind(position: Int) {
+            tribeMemberHolderViewState = tribeMembers.getOrNull(position) ?: let {
+                tribeMemberHolderViewState = null
+                return
+            }
+            disposable?.dispose()
+
```

Comment:
holderJobs and Disposables (from the image loader) should be iterated through here and cancelled prior to setting any data to the views